src/gatsby-node.js

```diff
@@ -10,23 +10,118 @@ exports.sourceNodes = (
   delete configOptions.plugins
 
   const sources = []
-
+  const dummyData = {
+    id: 'dummy',
+    appointment: '',
+    bio: '',
+    building: '',
+    consult_service: '',
+    experience: '',
+    grant_contract: '',
+    honor_award: '',
+    innovate_enterpreneur: '',
+    patent_invention: '',
+    pf_email: '',
+    pf_work_fax: '',
+    pf_first_name: '',
+    pf_last_name: '',
+    pf_work_phone: '',
+    pf_title: '',
+    pf_username: '',
+    website: '',
+    research: '',
+    room: '',
+    photo_base64: '',
+    notable_courses: '',
+    school: '',
+    service_university: [
+      {
+        org: '',
+        member_type: '',
+      },
+    ],
+    service_professional: [
+      {
+        title: '',
+        org: '',
+      },
+    ],
+    education: [
+      {
+        dty_comp: '',
+        deg: '',
+        degother: '',
+        school: '',
+        state: '',
+        country: '',
+        major: '',
+      },
+    ],
+    member: [
+      {
+        org: '',
+        status: '',
+        orgabbr: ''
+      },
+    ],
+    intellcont: [
+      {
+        contype: '',
+        contypeother: '',
+        journal_name: '',
+        pagenum: '',
+        status: '',
+        title: '',
+        volume: '',
+        publisher: '',
+        pubctyst: '',
+        issue: '',
+        dty_pub: '',
+        dty_acc: '',
+        dty_sub: '',
+        web_address: '',
+        intellcont_auth: {
+          faculty_name: '',
+          mname: '',
+          fname: '',
+          lname: '',
+        },
+      },
+    ],
+  }
+  
+  const dummyNodeContent = JSON.stringify(dummyData)
+  
+  const dummyNodeMeta = {
+    id: createNodeId(`PeopleFaculty-dummy`),
+    endpointId: "dummy",
+    parent: null,
+    children: [],
+    internal: {
+      type: `MultiApiSourcePeopleFaculty`,
+      content: dummyNodeContent,
+      contentDigest: createContentDigest(dummyData),
+    },
+  }
+  
+  const dummyNode = Object.assign({}, dummyData, dummyNodeMeta)
+  createNode(dummyNode)
   // Helper function that processes a result to match Gatsby's node structure
   const processResult = ({ result, endpoint, prefix }) => {
-    const nodeId = createNodeId(`${endpoint}-${result.id}`)
```

Comment:
Didn't this (replacing `result.id` with `result.pf_username`) already occur in `master`? It doesn't seem related to creating the dummy node, rather it was required for us to use this effectively even before finding that we need a dummy node. Commit `47dff446` specifically is when this change happened the first time. Based on https://github.com/danspratling/gatsby-source-multi-api/network I think you created this `Add-Dummy-Node` branch based on the upstream `master` instead of our fork's `master` or something, which explains the need to change this a second time, in which case you will get some nasty merge conflicts trying to merge this.

Reply:
Arrived at the same conclusion, good catch!

Since this is already merged, not much to do other than watch for regressions, but in the future you can stage the merge commit with `--no-commit --no-ff`, and undo it when you're done with `git merge --abort`.